### PR TITLE
Revert "Check flusher interface before calling Flush (#479)"

### PR DESCRIPTION
**Pull request by johanbrandhorst**
This reverts commit cf7d15a039114df2a0e3a18002705bb67fdd5167.
This introduced a recursive call to `Flush` in the proxy, which
will crash the application.

### CODE REVIEW

File: go/grpcweb/grpc_web_response.go [hunk 1]
```diff
@@ -95,7 +90,7 @@ func (w *grpcWebResponse) finishRequest(req *http.Request) {
 		w.copyTrailersToPayload()
 	} else {
 		w.WriteHeader(http.StatusOK)
-		w.Flush()
```
Comment (jonny-improbable) [inline go/grpcweb/grpc_web_response.go]: @johanbrandhorst whilst we wait for CI to go green on this PR...

...isn't this the root cause of the issue? Shouldn't this be replaced with a safe up-cast on `w.wrapped` before calling `Flush()`?

**Comment by jonny-improbable** [comment]
Given this failure has had an impact on consumers, it implies we are missing a test. @light0x, would you be in a position to help us contribute a test-case which would help us catch this regression in future, please?

**Comment by johanbrandhorst** [comment]
Rerunning failed tests

File: go/grpcweb/grpc_web_response.go (hunk 1 above)
Comment (johanbrandhorst) [reply go/grpcweb/grpc_web_response.go]: I haven't looked at it in detail, lets discuss after merging the fix.

**Comment by jonny-improbable** [comment]
Given that the 2 failures occurred on older versions of Chrome I am happy to chalk the failure down to BrowerStack flakiness and will use Admin powers to merge.

**Pull request merged**